### PR TITLE
Timetable widget configuration

### PR DESCRIPTION
**Pull request by sebmarste**
## Screenshot
screenshot_20170611-235157
screenshot_20170617-142138

#### config: 
- group lectures by lecture name
- save blacklist for widget
- filter by blacklist
#### widget:
- add intent to open roomfinder

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/adapters/LectureListSelectionAdapter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public View newView(Context context, Cursor cursor, ViewGroup parent) {
+        return LayoutInflater.from(context).inflate(R.layout.list_timetable_configure_item, parent, false);
+    }
+
+    @Override
+    public void bindView(View view, Context context, Cursor cursor) {
+        CheckBox checkBox = (CheckBox) view.findViewById(R.id.timetable_configure_item);
+        checkBox.setChecked(cursor.getInt(cursor.getColumnIndex("is_on_blacklist")) == 0);
+        checkBox.setText(cursor.getString(cursor.getColumnIndex("title")));
+        checkBox.setOnCheckedChangeListener(this);
```
Comment (sebmarste) [inline app/src/main/java/de/tum/in/tumcampusapp/adapters/LectureListSelectionAdapter.java]: maybe use setTag to save the lecture "id" and then simplify the lecure title for the text view

File: app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java [hunk 2]
```diff
@@ -158,18 +161,27 @@ public Cursor getFromDbForDate(Date date) {
...
         String to = Utils.getDateTimeString(calendar.getTime());
 
         List<IntegratedCalendarEvent> calendarEvents = new ArrayList<>();
-        Cursor cursor = db.rawQuery("SELECT * FROM calendar WHERE dtend BETWEEN ? AND ? AND status!='CANCEL' ORDER BY dtstart ASC", new String[]{from, to});
+        Cursor cursor;
+        if (widgetId >= 0) {
+            cursor = db.rawQuery("SELECT * FROM calendar c WHERE dtend BETWEEN ? AND ? AND status!='CANCEL' " +
+                    "AND NOT EXISTS (SELECT * FROM widgets_timetable_blacklist WHERE widget_id=? AND lecture_title=c.title) " +
+                    "ORDER BY dtstart ASC", new String[]{from, to, String.valueOf(widgetId)});
+        } else {
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java]: Why is this check necessary here? Can the `widgetId` ever be <0?

File: app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java [hunk 3]
```diff
@@ -202,6 +214,42 @@ public boolean hasLectures() {
...
+     */
+    public void addLectureToBlacklist(int widgetId, String lecture) {
+        ContentValues values = new ContentValues();
+        values.put("widget_id", widgetId);
+        values.put("lecture_title", lecture);
+        db.replace("widgets_timetable_blacklist", null, values);
+    }
+
+    /**
+     * Remove a lecture to the blacklist of a widget
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java]: Remove ~to~ from the blacklist

File: app/src/main/java/de/tum/in/tumcampusapp/widgets/TimetableWidgetConfigureActivity.java [hunk 4]
```diff
@@ -1,7 +1,80 @@
...
+
+    /**
+     * Setup cancel and back action
+     *
+     * @param item the menu item which has been pressed (or activated)
+     */
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/widgets/TimetableWidgetConfigureActivity.java]: A `if` would be simpler here

File: app/src/main/res/drawable/ic_check.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (pfent) [inline app/src/main/res/drawable/ic_check.xml]: I think, we already have the same file as `ic_dome.xml`. Feel free to update / rename the old file to match the material icon names

File: app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java (hunk 2 above)
Comment (sebmarste) [inline app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java]: this was from the first version of the widget (without filtering) -> removed

File: app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java (hunk 3 above)
Comment (sebmarste) [inline app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java]: done

File: app/src/main/java/de/tum/in/tumcampusapp/widgets/TimetableWidgetConfigureActivity.java (hunk 4 above)
Comment (sebmarste) [inline app/src/main/java/de/tum/in/tumcampusapp/widgets/TimetableWidgetConfigureActivity.java]: done

File: app/src/main/res/drawable/ic_check.xml (hunk 5 above)
Comment (sebmarste) [inline app/src/main/res/drawable/ic_check.xml]: i use ic_done now

**Comment by sebmarste** [comment]
@pfent yes the PR is ready to merge and from my side the widgets are done 👍

**Pull request merged**